tools/fastdds/discovery/fast_daemon/process_handler/process_handler.py

```diff
@@ -0,0 +1,98 @@
+import os
+import select
+import signal
+import subprocess
+import threading
+
+class ProcessHandler:
+    def __init__(self, toolpath: str):
+        self.processes = {}  # Dictionary to store process information
+        # Path to the tool to be executed. It is stored in the object to avoid
+        # passing it as an argument in every call and to enable daemon calls between
+        # host and docker containers with --net=host option.
+        self.toolpath = toolpath
+        self._lock = threading.RLock()
+
+    def __del__(self):
+        # Ensure all processes are stopped when the object is deleted
+        with self._lock:
+            for domain in list(self.processes.keys()):
+                self.stop_process(domain, 1)
+
+    def _get_signal(self, sig: int):
+        if sig == 0:
+            return signal.SIGINT
+        elif sig == 1:
+            return signal.SIGTERM
+        elif sig == 2:
+            return signal.SIGKILL
+
+    def run_process_nb(self, domain: int, command: list):
+        """ Used for starting new servers. Commands 'start' and 'auto'."""
+        with self._lock:
+            if domain in self.processes:
+                return f"Discovery server for Domain '{domain}' is already running."
+            # Start a new subprocess in a non-blocking way
+            command.insert(0, self.toolpath)
+            process = subprocess.Popen(command,
+                                       stdout=subprocess.PIPE,
+                                       stderr=subprocess.PIPE,
+                                       preexec_fn=os.setsid)
+            # Check output to see if the process started correctly or it issued an error
+            try:
+                streams, _, _ = select.select([process.stdout, process.stderr], [], [], 0.3)
+                output = ''
+                if streams:
+                    for stream in streams:
+                        output = os.read(stream.fileno(), 1024).decode('utf-8')
```

Comment:
We agreed on processing here the `stdout` only since its the intersting one to know if the server was sucessfully created
```suggestion
                output = os.read(process.stdout.fileno(), 1024).decode('utf-8')
```